lib/foreman_maintain/concerns/base_database.rb

```diff
@@ -14,12 +14,15 @@ def query_csv(sql, config = configuration)
       end
 
       def psql(query, config = configuration)
-        execute("PGPASSWORD='#{config[%(password)]}' #{psql_db_connection_str(config)}",
-                :stdin => query)
+        if ping(config)
+          execute(psql_connection_string(config), :stdin => query)
+        else
+          raise_service_error
+        end
       end
 
       def ping(config = configuration)
-        psql('SELECT 1 as ping', config)
+        execute?(psql_connection_string(config), :stdin => 'SELECT 1 as ping')
```

Comment:
One last nitpick: `connection_string` phrase is connected to a special case (or at least for me, and I can imaging others having similar connotation) of passing to some driver to connect to db. In this case, I would rather go for `psql_command_string` or simply just `psql_command`